### PR TITLE
set useNativeDriver to true in request-button animation

### PR DESCRIPTION
**Pull request by rasom**
`useNativeDriver: true` significantly reduces the number of messages from/to RN bridge, each animation caused ~60 messages per second. 

https://facebook.github.io/react-native/blog/2017/02/14/using-native-driver-for-animated.html

related to #2963 

status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

**Comment by rasom** [comment]
On galaxy note 4:
1) when chat has 10 plain messages it uses **~45% cpu**
2) 10 plain messages + 4 requests **~96-101% cpu** when `useNativeDriver = false`
3) 10 plain messages + 4 requests **~50-55% cpu** when `useNativeDriver = true`

**Comment by oskarth** [comment]
Yes! I saw this option in some RN perf article I read recently, awesome to see we can easily use it to get real perf gains :)

**Review by oskarth — APPROVED** [review]

**Review by yenda — APPROVED** [review]

**Comment by asemiankevich** [comment]
@rasom me and @antdanchenko have made some perf measurement for this PR. 
- Perfomance test was run against release build and current branch
- There was used real device LG Google Nexus 5X Android 6.0.1.
- Scenario tested: 10 plain text messages and several request messages in 1x1 chat

Unfortunately I dont see any significant CPU usage decrease , please check the charts.

image

image

Please let me know if it helps at least a bit. As well if I misunderstood the scenario or anything. Blue line is current branch and yellow is release build.

**Comment by rasom** [comment]
It's about 15% cpu per one request message when chat is in standby (when it's opened for 30 seconds or something), though 
1) you can miss this difference because of other cpu intensive tasks, random fluctuations, whatever
2) i really have difficulties with reading these graphs (especially the first one)
3) hence don't see any reason to compare performance before/after using exactly this metric 
¯\\_(ツ)_/¯

**Comment by asemiankevich** [comment]
Well I don't see any other way to test it)

**Comment by yenda** [comment]
@asemiankevich out of curiosity what are you using to make the charts ?

Also if you didn't find any regression let's merge this.

**Comment by oskarth** [comment]
Great graphs and tests @asemiankevich and @antdanchenko!

@rasom: 
> you can miss this difference because of other cpu intensive tasks, random fluctuations, whatever

Easy fix: re-run specific scenario 5-10 times and take average.

> i really have difficulties with reading these graphs (especially the first one)

What would make it easier? I'm sure a table can also be produced quite easily.

> hence don't see any reason to compare performance before/after using exactly this metric

Can you define a procedure that QA can auto test? I.e. idle state or whatever.

---

Agree with @yenda that we should merge, assuming no regressions,  even if no direct measurable impact, as there is local CPU impact and this is the right way to do this.

**Comment by rasom** [comment]
Let's wait with merge, i will explain @asemiankevich how to test exactly this case. I just need some time, will do this later today

**Comment by rasom** [comment]
> Can you define a procedure that QA can auto test? I.e. idle state or whatever.

The way how you can test cpu usage in this case

1. install app on device A 
2. create an account
3. send 3 request messages from another device to device A
4. restart app
5. check cpu usage using `adb shell dumpsys cpuinfo | grep "im.status.ethereum"`, output will be like
`69% 14261/im.status.ethereum: 55% user + 14% kernel / faults: 18106 minor`
6. when we just logged in, as far as i understand, app tries to find peers and probably does other black magic, but the fact is that cpu usage increases up to 370-380% on galaxy note 4. We need to wait 2-3 minutes before this craziness ends.
7. after 2-3 minutes we can check cpu again, on galaxy note 4 it's 33-40% when chats list is opened.
8. open chat with 3 requests messages
9. wait a minute (probably we need less time, but the point is that changes are not reflected in `dumpsys cpuinfo` immediately) 
10. check cpu few times again using the same `adb shell dumpsys cpuinfo | grep "im.status.ethereum"`, probably with 20 seconds interval or so.
11. restart app and repeat 5-10 steps to make sure that cpu usage numbers can be reproduced
12. note these numbers

So when i did this using release build cpu usage in chat with 3 requests messages was in interval `69-78%`. This branch showed `45-46%`.

I'm not sure that we need to cover exactly this scenario with auto tests. For sure we need to wait these 2-3 minutes after login, because such small cpu improvements will be lost in initial spikes.

**Comment by rasom** [comment]
> What would make it easier? I'm sure a table can also be produced quite easily.

I'm not sure that table such data will be more useful. What exactly can you say from this graph? Does it show any regression, improvement or performance is the same? For me it's super hard to parse visually what is happening there.

**Comment by Serhy** [comment]
The `dumpsys cpuinfo` reflects the changes in my case too slow - delay maybe > 2 minutes or so. CPU usage I took from `adb shell top` response with the steps offered above, but a bit modified to get average and more precise figures.

I've used the `adb shell top -d 1 -n 120 | grep "im.status.ethereum"`
It starts a loop and each second checks the CPU usage (along with other data for current process filtered with grep) of the application during next two minutes. 

###Verification steps flow:

1) Install app on device A (Android 7.0 on Samsung J7 (2017) in my case)
2) Install app on device B (iPhoneX)
3) Create accounts on both devices. Send three /request messages (/request 0) from device B to device A.
4) Close the App on DeviceA
5) Execute in terminal `adb shell top -d 1 -n 120 | grep "im.status.ethereum"`
6) Open Status App on DeviceA
7) Log in to account. Once you logged in - keep an eye on CPU usage in terminal what we started in 5th step. Right after log in on my device CPU usage ~60-100% as app tries to find peers and probably does other black magic (what @rasom talks about in https://github.com/status-im/status-react/pull/3072#issuecomment-358455311)
8) At some point (in ~30 seconds after login) you may see the CPU usage dropped to 0-5% (which means app finished to look for peers and sync). Tap on 1-1 chat with the user where you have 3 /request messages and stay on this page till the `adb shell top -d 1 -n 120 | grep "im.status.ethereum"` finished.
9) Keep the results and take average CPU usage app had in previous 8th item above.
10) Repeat 4th-9th steps 5-10 times and get the average CPU usage from several runs.

Below are two files with the CPU% data from several runs per two app versions:
The data hightlighted in orange colour - is the CPU% when user taps on 1-1 chat with three /request messages. And below it in green - is the average of CPU from what highligted in orange (so average CPU usage on Status App when user stays on 1-1 chat screen)

* Release 0.9.13 (from 19th of January) App version from Android 7.0 on Samsung J7 (2017) : 
[Release0913_CPU_usage.pdf](https://github.com/status-im/status-react/files/1646562/Release0913_CPU_usage.pdf)

* 162fea2b+ App version from Android 7.0 on Samsung J7 (2017): 
[162fea2b_average_CPU_usage.pdf](https://github.com/status-im/status-react/files/1646568/162fea2b_average_CPU_usage.pdf)

To finalise the result: 
- Samsung J7 (2017) with Android 7.0. Release 0.9.13 (from 19th of January 2017) app version.  Average CPU% usage, when user stays on 1-1 chat with three request messages - is **~14.23%**
- Samsung J7 (2017) with Android 7.0. 162fea2b+ app version. 
Average CPU% usage, when user stays on 1-1 chat with three request messages - is **~11.44%**

With `useNativeDriver: true` CPU usage indeed reduced in ~20% (and on ~30% CPU usage reduced on galaxy note 4, according to @rasom tests). 

Additionally basic regression performed on Android 7.0 and iPhoneX (iOS 11.2.1). Upgrade from 0.9.12 to 162fea2b+ tested too. No bugs found. 

Based on these results I think we can merge it!
